### PR TITLE
Feature: Add CSS build

### LINKED ISSUE #24: Add css build (webpack) into project.
Add some CSS somewhere ;)
- [ ] prod build
- [ ] hot loading

/cc @alexlande

### PR DESCRIPTION
**Pull request by alexlande**
This was a pain, which makes me very happy that we'll have a centrally located version of it in this repo.

This adds a very simple CSS build system (not including any preprocessors, as those are included with additional loaders but won't affect the infrastructure otherwise), and pulls in Bootstrap with it instead of over CDN.

To pull in Bootstrap, I had to include support for image and font files, which I would have done in a separate PR anyway so it worked out.

I'll leave additional comments on any hairy parts of this that may need extra attention.

Resolves #24 

/cc @ryan-roemer

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -35,6 +35,7 @@
     "start": "node server/index.js",
     "server": "nodemon --watch client --watch server --watch templates --ext js,jsx server/index.js",
     "server-dev": "nodemon --watch client --watch server --watch templates --ext js,jsx server/index-dev.js",
+    "server-hot": "nodemon --watch client --watch server --watch templates --ext js,jsx server/index-hot.js",
```
Comment (alexlande) [inline package.json:38]: I needed a hook to add a `WEBPACK_HOT` environmental variable. I followed the pattern that we've got for dev, but am not in love with the repetition here.

File: package.json [hunk 2]
```diff
@@ -72,11 +73,17 @@
     "bluebird": "^2.9.34",
     "clean-webpack-plugin": "^0.1.3",
     "compression": "^1.5.2",
+    "css-loader": "^0.19.0",
+    "es6-promise": "^3.0.2",
```
Comment (alexlande) [inline package.json:78]: I had nvm set to 0.10 when I started this, and discovered that `css-loader` requires a promise polyfill now. We have fun here in JavaScript-land.

File: server/index.js [hunk 3]
```diff
@@ -11,6 +11,8 @@ var RENDER_SS = true;
 
 // Hooks / polyfills
 require("babel/register");
+require.extensions[".css"] = function () { return null; };
```
Comment (alexlande) [inline server/index.js:15]: On the server, we need to prevent CSS requires from attempting to do things.

File: server/index.js [hunk 4]
```diff
@@ -75,10 +77,11 @@ app.indexRoute = function (root) {
   var fluxMiddleware = mid.flux.actions(Page); // Instance actions.
 
   app.use(root, [fluxMiddleware], function (req, res) {
-    /*eslint max-statements:[2,20]*/
+    /*eslint max-statements:[2,25]*/
```
Comment (alexlande) [inline server/index.js:81]: I felt bad about this, but given that it already included an override from the default `15` max-statements, I didn't feel _that_ bad.

File: test/func/mocha.opts [hunk 5]
```diff
@@ -1,3 +1,3 @@
 --require test/func/setup.js
 --recursive
---timeout 10000
+--timeout 20000
```
Comment (alexlande) [inline test/func/mocha.opts:3]: I observed this integrating the functional testing suite in another project as well. The 10000 timeout limit didn't cut it anymore. I wasn't sure if the right thing was to try to narrow it down to the particular cases that were pushing it over, or just to set it higher in general.

In principle, setting a high timeout limit at the beginning feels more reasonable to me than sprinkling in overrides in tests that happen to be slow, so I went with this for now.

File: server/index.js (hunk 3 above)
Comment (ryan-roemer) [inline server/index.js:15]: Can you add a code comment here to that effect?

File: server/index.js (hunk 4 above)
Comment (ryan-roemer) [inline server/index.js:81]: No worries. In a real app this should be decomposed ;)

**Comment by ryan-roemer** [comment]
@alexlande -- You've got some CI fixin' I think:

```
RROR in ./client/components/page.jsx
Module not found: Error: Cannot resolve module 'bootstrap/dist/css/bootstrap-theme.css' in C:\projects\converter-react\client\components
 @ ./client/components/page.jsx 50:0-49
Child extract-text-webpack-plugin:

29 09 2015 22:18:37.444:INFO [karma]: Karma v0.13.10 server started at http://localhost:9999/
29 09 2015 22:18:37.444:INFO [launcher]: Starting browser PhantomJS
29 09 2015 22:18:37.475:INFO [launcher]: Starting browser IE
29 09 2015 22:18:40.248:INFO [PhantomJS 1.9.8 (Windows 8 0.0.0)]: Connected on socket BO2-OPq0tIui2EqiAAAA with id 53817634
PhantomJS 1.9.8 (Windows 8 0.0.0) ERROR
  Error: Cannot find module "bootstrap/dist/css/bootstrap.css"
  at C:/projects/converter-react/test/client/main.js:33949
29 09 2015 22:18:42.586:INFO [IE 11.0.0 (Windows 8.1 0.0.0)]: Connected on socket EunoGyGMDbThFABAAAAB with id 29960317
IE 11.0.0 (Windows 8.1 0.0.0) ERROR
  Cannot find module "bootstrap/dist/css/bootstrap.css"
  at C:/projects/converter-react/test/client/main.js:33949

PhantomJS 1.9.8 (Windows 8 0.0.0): Executed 0 of 0 ERROR (0.318 secs / 0 secs)
IE 11.0.0 (Windows 8.1 0.0.0): Executed 0 of 0 ERROR (0.383 secs / 0 secs)
```

**Comment by ryan-roemer** [comment]
Travis too:

```

ERROR in ./components/page.jsx
Module not found: Error: Cannot resolve module 'bootstrap/dist/css/bootstrap.css' in /home/travis/build/FormidableLabs/converter-react/client/components
 @ ./components/page.jsx 48:0-43
ERROR in ./components/page.jsx
Module not found: Error: Cannot resolve module 'bootstrap/dist/css/bootstrap-theme.css' in /home/travis/build/FormidableLabs/converter-react/client/components
 @ ./components/page.jsx 50:0-49
Child extract-text-webpack-plugin:
        + 2 hidden modules
```

**Comment by alexlande** [comment]
Blergh. That's interesting...

**Comment by ryan-roemer** [comment]
@alexlande -- Generally looking great! Can you fix CI and ping for re-review?

**Comment by alexlande** [comment]
I (embarrassingly) somehow did not `--save` when I installed bootstrap.

File: package.json (hunk 1 above)
Comment (ryan-roemer) [inline package.json:38]: @alexlande -- This repo is also fully windows compatible. The usual `FOO=bar my-command.js` breaks windows.

**Comment by ryan-roemer** [comment]
@alexlande -- LGTM.

I've tried to adjust `coveralls` failing the build (which we don't want -- _we_ should control that from within `istanbul` and our infrastructure in the repo). Ping me if the `coveralls/coveralls` service continues failing us. In the meantime, ignore this failure for this PR.

Nice work and thanks!

**Pull request merged**